### ISSUE TITLE
Client fails after resuming laptop

### ISSUE THREAD

**Issue by pzwahlen** [nextcloud/desktop#489]
I'm testing the latest 2.5 since a couple weeks, currently running "Nextcloud-2.5.0.0-Debug-20180710.exe" on Windows 10 Enterprise x64 1803. It's running against 13.0.4 on the server side.

When laptop resumes from either sleep or hibernate, client shows a red cross in the tray icon. Stopping and restarting the client solves the problem.

Nextcloud_sync.log and .owncloudsync.log do not show anything useful.

I'm also referencing #324, even though it relates to the stable 2.3 build which has been working fine for me, up to the latest 2.3.3.1.

Finally, I'm not using any of the new features (like E2E encryption).

Please let me know how to extract useful logs, happy to do so.

**Comment by pzwahlen** [comment]
While trying to recover from this situation, I usually either exit the client or double-click on it. This generate the following crash logs:

```
Faulting application name: nextcloud.exe, version: 2.5.0.0, time stamp: 0x5b4406cf
Faulting module name: Qt5Networkd.dll, version: 5.9.5.0, time stamp: 0x5ac4db27
Exception code: 0xc0000005
Fault offset: 0x0000000000065d33
Faulting process id: 0x2638
Faulting application start time: 0x01d419b7f365df85
Faulting application path: C:\Program Files (x86)\Nextcloud\nextcloud.exe
Faulting module path: C:\Program Files (x86)\Nextcloud\Qt5Networkd.dll
Report Id: 4720aaad-0a0e-4ccd-bac6-ca3f15ebe840
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Faulting application name: nextcloud.exe, version: 2.5.0.0, time stamp: 0x5b4406cf
Faulting module name: Qt5WebEngineCored.dll, version: 5.9.5.0, time stamp: 0x5acd504d
Exception code: 0x80000003
Fault offset: 0x000000000176854b
Faulting process id: 0xc98
Faulting application start time: 0x01d41a04a3a5e1db
Faulting application path: C:\Program Files (x86)\Nextcloud\nextcloud.exe
Faulting module path: C:\Program Files (x86)\Nextcloud\Qt5WebEngineCored.dll
Report Id: cb1062bd-23be-411c-b6c6-89eab0b86e09
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by rednaxela-work** [comment]
Have the same issue on Windows 7 64 bit.

After resuming hibernation the client shows a red cross and red error messages inside the window. 

Closing the client ends in Windows failing to end the process and another error message.

Only after this I can restart the client.

**Comment by pzwahlen** [comment]
Yesterday I made a new test with the latest nightly build for Windows (Nextcloud-2.5.0.61457-daily-20181009.exe).

This issue is still present and after wake-up, the client was showing a red-cross in the tray area. Left-clicking would open the GUI and a red banner mentioned a socket timeout.

Trying to right-click in order to reach the menu and exit the client didn't work. The icon simply disappeared which I think is a sign of the client either crashing or at least exiting in a strange way.

I still couldn't find any useful log.

Server is now running 14.0.1

#324 looks similar but on the "stable" client, which for me is working like a charm and perfectly survives sleep/resume.

Any way to help further troubleshoot this issue ?

**Comment by pzwahlen** [comment]
A couple days ago I installed 2.5.0-rc1 and so far I haven't seen the red cross. Fingers crossed.

CCing @camilasan @rednaxela-work

**Comment by jauling** [comment]
The NextCloud Desktop client for MacOS also stops working when my laptop is resumed from sleep. Even if I pause and resume the sync, it doesn't recover. It looks like I have to manually go into the settings, click on the `Account` button, and click on `Log in`. Shouldn't it retry?

My Nextcloud server is running 13.0.8.

```
Version 2.5.0daily (build 20181112)
Built from Git revision 371001 on Nov 12 2018, 01:03:46 using Qt 5.9.2, OpenSSL 1.0.2m 2 Nov 2017
```

**Comment by pzwahlen** [comment]
The situation has greatly improved with 2.5. I sometime had the client crashing on me when trying to open the UI (with a QT app crash event log), however since 2.5.1 this seems to be fixed. I'll open a new issue if I can reproduce this crash.

Thanks for the hard work!

**Comment by tinola** [comment]
Version 2.5.1 (build 20181204) - same issue - red cross after resume from sleep :( Restart client is necessary. Windows 10 x64 (1803)

**Comment by fzieris** [comment]
Same here (using Windows 10). It is not limited to hibernation, though. It's the same issue when I connect to or disconnect from a VPN network (using Cisco AnyConnect, in case that matters to anyone). It looks like any change to the network adapter throws the client off the rails.

**Comment by Exception0815** [comment]
Same issue with Version 2.5.1final (build 20181204), Win10 1709 German.
By the way I have Comodo Firewall installed but the logs are not showing any entries.

NC client log:
QNetworkReply::NetworkError(UnknownNetworkError)
[OCC::WebFlowCredentials::stillValid 	"Der Zugriff auf das Netzwerk ist nicht gestattet."
[OCC::PropfindJob::finished 	PROPFIND of QUrl("***") FINISHED WITH STATUS "UnknownNetworkError Der Zugriff auf das Netzwerk ist nicht gestattet."
[OCC::PropfindJob::finished 	*not* successful, http result code is 0 ""

**Comment by Niondir** [comment]
Same here: 2.5.1 on Windows 10. This is pretty annoying.

Is there any workaround that does not include restarting Nextcloud?

**Comment by rwilliamsit** [comment]
I ended up downgrading my client to the one that was out before 2.5... I think it was released Dec 2017? I’m not at my desktop right now.  I got tired of this bug and also another regression I noticed where the 2.5.x client tries to sync files that are still open for write which results in an error that it failed to sync...

**Comment by Haplo164** [comment]
Same issues, It seems to happen when windows stops network services.

**Comment by Exception0815** [comment]
As a workaround create a Powershell script with following two lines, and run it automatically everytime you wake your PC by using a scheduled task. It will then restart the Nextcloud application:
`Stop-Process -name nextcloud`
`Start-Process -FilePath "C:\Program Files (x86)\Nextcloud\nextcloud.exe"`

Create then a new scheduled task, enter following XML code as a trigger:
`<QueryList>
  <Query Id="0" Path="System">
    <Select Path="System">*[System[Provider[@Name='Microsoft-Windows-Power-Troubleshooter'] and (Level=4 or Level=0) and (EventID=1)]]</Select>
  </Query>
</QueryList>`

To run a PS script as a scheduled task see [https://community.spiceworks.com/how_to/17736-run-powershell-scripts-from-task-scheduler](url)

**Comment by noja11** [comment]
I'm also experiencing this issue.  When the laptop system sleeps/hibernates, the client crashes.  Running Version 2.5.1final (build 20181204). 

After a few tests however, I can see that the service has stayed up once I un-ticked "Launch on system startup" under the settings tab.  Of course I have to remember to launch the client on full reboot, but as this is a laptop, that happens less frequently than sleep/hibernate.

**Comment by pgpbj** [comment]
Same issue for me.  Sleeping/Hibernating Windows 10 will cause the 2.5.1 client to fail after resuming.  Restarting client will fix the failure.

Trivial to reproduce, yet still at triage stage after 7 months?

Will experiment with the suggested workarounds above...

**Comment by Niondir** [comment]
Just tested with "start on boot" disabled. Does not help with my machine.

**Comment by noja11** [comment]
> 
> 
> Just tested with "start on boot" disabled. Does not help with my machine.

Sad that's not working.  I've noticed on my end, that if I just put my laptop to Sleep, it'll wakeup and the client will work just fine.  However, if it ends up closed for too long and moves into Hibernate, I'm still getting the crash.

**Comment by jprokosch** [comment]
Same issue for me in Windows 10 and Nextcloud client 2.5.1
Annoying and systematic problem !
Hope to get a fix soon as possible !

**Comment by pgpbj** [comment]
I downgraded to version 2.3.3.1 (from https://download.nextcloud.com/desktop/releases/Windows/) and, so far, the problem seems to be gone.

**Comment by NiklasOlofsson** [comment]
Same here. Windows 10. My main computer waking from sleep, white X, red background on NextCloud client icon.

Version 2.5.1final (build 20181204).
Git revision b37cbe den Dec 4 2018, 13:30:31 med Qt 5.11.1, OpenSSL 1.0.1h 5 Jun 2014

**Comment by ThomasCr** [comment]
I added some more infos on https://github.com/nextcloud/desktop/issues/324#issuecomment-483336116

**Comment by elforesto** [comment]
I manually updated to 2.5.2 and the client now behaves as expected on resuming from sleep or hibernate.

**Comment by ThomGr** [comment]
Still see almost always a crash of the client (2.5.2, build 20190319, git 56c905, Windows 10 1903) when resuming from hibernate on a Lenovo T480s. Any progress on this? I would be happy to send debug information or a crash dump.

**Comment by misch7** [comment]
Have you also tried our latest release 2.5.3 ?
https://download.nextcloud.com/desktop/releases/Windows/Nextcloud-2.5.3-setup.exe

And it this still happens with 2.5.3 would you try one of our current daily builds, e.g.:
https://download.nextcloud.com/desktop/daily/Windows/Nextcloud-2.5.3.5732-daily-20190820-Release.exe

Since 2019-08-18 we're compiling against Qt 5.12.4 so there's a chance it might have fixed the issue accidently ;-)

I'm curious how it turns out.

**Comment by ThomGr** [comment]
interesting, 2.5.3 seems to solve the issue. I will monitor it and try a daily build should the problem come back.
BTW, I checked in the client and it did not offer any updates, so I did not look on the website. Thanks for mentioning the latest version!

**Comment by ThomGr** [comment]
in contrast to my earlier post, 2.5.3 does _not_ solve the issue. The client keeps crashing after resuming from hibernate state, not from power saving state.

**Comment by misch7** [comment]
Could you also try our recent 2.6.0 release build?

https://download.nextcloud.com/desktop/releases/Windows/Nextcloud-2.6.0-setup.exe

Qt 5.12.4 had a lot of issues with UI and crashes so perhaps our upgrade to Qt 5.12.5 in the last release solves it - *fingers crossed*

**Comment by ThomGr** [comment]
no change with 2.6.0

**Comment by ice954pw** [comment]
Same Issue with 2.6 as well

**Comment by caric87** [comment]
same issue on my end too. 2.6 on Windows 7

**Comment by ice954pw** [comment]
Fixed on latest build, 2.6.1

**Comment by ThomGr** [comment]
installed 2.6.1 today, but the issue is not fixed for me

**Comment by adaigre** [comment]
I'm on archlinux and my client fails after sleep.
I don't have a systray/notification icon (I'm using Gnome) but the client only seems to freeze.
After a few seconds, the client close himself.

I've also got a Windows machine who seems to have the same problem.
My Nextcloud installation was recently moved to a new server but no change ...
I'm using Debian 10 with nginx, php-fpm (7.3) and MariaDB.

`[OCC::FolderMan::scheduleFolderNext 	Schedule folder  "1"  to sync! Front-of-queue.
[OCC::SocketApi::slotUpdateFolderView 	Not sending UPDATE_VIEW for "1" because status() is 1
[OCC::ownCloudGui::slotSyncStateChange 	Sync state changed for folder  "https://my-nextcloud.fr/remote.php/dav/files/alexandre/" :  "Not yet Started"
[OCC::AccessManager::createRequest 	2 "" "https://my-nextcloud.fr/ocs/v2.php/apps/notifications/api/v2/notifications?format=json" has X-Request-ID "696134f4-1c14-46dc-a2ce-3157addcaf0a"
[OCC::AbstractNetworkJob::start 	OCC::JsonApiJob created for "https://my-nextcloud.fr" + "ocs/v2.php/apps/notifications/api/v2/notifications" "OCC::ServerNotificationHandler"
[OCC::FolderMan::startScheduledSyncSoon 	Starting the next scheduled sync in 0 seconds
[OCC::SocketListener::sendMessage 	Sending SocketAPI message --> "STATUS:OK:/home/alexandre/Nextcloud" to QLocalSocket(0x557873dbcac0)
[OCC::SocketListener::sendMessage 	Sending SocketAPI message --> "UPDATE_VIEW:/home/alexandre/Nextcloud" to QLocalSocket(0x557873dbcac0)
[OCC::ownCloudGui::slotSyncStateChange 	Sync state changed for folder  "https://my-nextcloud.fr/remote.php/dav/files/alexandre/" :  "SyncPrepare"
[OCC::Folder::startSync 	*** Start syncing  "https://my-nextcloud.fr/remote.php/dav/files/alexandre/"  - client version 2.6.1git
[OCC::Folder::startSync 	Allowing local discovery to read from the database
[OCC::FolderMan::slotFolderSyncStarted 	>========== Sync started for folder [Nextcloud] of account [alexandre@my-nextcloud.fr] with remote [https://my-nextcloud.fr/remote.php/dav/files/alexandre/]
[OCC::SyncEngine::startSync 	There are 181011492864 bytes available at "/home/alexandre/Nextcloud/"
[OCC::SyncEngine::startSync 	Sync with existing sync journal
[OCC::SyncEngine::startSync 	"Using Qt 5.13.2 SSL library OpenSSL 1.1.1d  10 Sep 2019 on Arch Linux"
[OCC::SyncEngine::startSync 	Using Selective Sync
[OCC::SyncEngine::startSync 	#### Discovery start ####################################################
[OCC::SyncEngine::startSync 	Server "17.0.1.1" Using HTTP/2
[csync_memstat_check 	Memory: 1329640K total size, 161272K resident, 85300K shared
[csync_update 	## Starting local discovery ##
[fill_tree_from_db 	8700 entries read below path  from db.
[csync_update 	Update detection for local replica took 0.066 seconds walking 8700 files
[csync_memstat_check 	Memory: 1329640K total size, 161284K resident, 85300K shared
[csync_update 	## Starting remote discovery ##
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://my-nextcloud.fr/remote.php/dav/files/alexandre/" has X-Request-ID "5c34aaa8-dd69-4ee7-bbc6-9eb65e14af32"
[OCC::AbstractNetworkJob::start 	OCC::LsColJob created for "https://my-nextcloud.fr" + "" "OCC::DiscoverySingleDirectoryJob"
[OCC::SocketApi::onLostConnection 	Lost connection  QLocalSocket(0x557873dbcac0)`

**Comment by fkbreitl** [comment]
Same problem with version 2.6.4 from https://portapps.io/app/nextcloud-portable/ .

**Comment by sanchobear** [comment]
Sane problem with version 2.6.4stable-Win64 on Win10 1909. Need to restart after resume from power saving.
Eventlog: (sorry German)
Name der fehlerhaften Anwendung: nextcloud.exe, Version: 2.6.4.15215, Zeitstempel: 0x5e5eb93a
Name des fehlerhaften Moduls: Qt5Core.dll, Version: 5.12.5.0, Zeitstempel: 0x5d6ebdfc
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000001c9cc3
ID des fehlerhaften Prozesses: 0x19a8
Startzeit der fehlerhaften Anwendung: 0x01d600f941800c10
Pfad der fehlerhaften Anwendung: C:\Program Files\Nextcloud\nextcloud.exe
Pfad des fehlerhaften Moduls: C:\Program Files\Nextcloud\Qt5Core.dll
Berichtskennung: e1f373d2-3482-44fb-87a5-0ebd827545a2
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist: 

See WIndows crash report (WER file renamed)
[Report..wer.txt](https://github.com/nextcloud/desktop/files/4369392/Report.wer.txt)

**Comment by gregholst** [comment]
I also need to restart the Nextcloud client on MacOS since several versions every time after sleep mode was used. Current version: Version 2.6.4stable (build 20200303). MacOS Catalina

I ran the client with the log option. See the last lines of the log and also what I received on the terminal:

`
[_csync_detect_update 	file: documentation/react_dev_1.docx, instruction: INSTRUCTION_NONE <<=
[csync_ftw 	 <= Closing walk for /Users/greg/Nextcloud/documentation with read_from_db 0
[_csync_detect_update 	.DS_Store excluded  (2)
[_csync_detect_update 	Database entry found for greg, compare: 1579429545 <-> 1579165057, etag:  <-> 5e2025813af71, inode: 12890516466 <-> 12890516466, size: 544 <-> 0, perms: 0 <-> fd, checksum:  <->  , ignore: 0,  e2e: 
[unknown 	QEventDispatcherUNIXPrivate(): Can not continue without a thread pipe
[OCC::ActivityListModel::clearNotifications 	Clear the notifications
[OCC::ActivityListModel::addNotificationToActivityList 	Notification successfully added to the notification list:  "Update to Nextcloud 18.0.3 is available."

greg@Etna private %  /Applications/nextcloud.app/Contents/MacOS/nextcloud --logfile nextcloud.log
QThreadPipe: Unable to create pipe: Too many open files
zsh: abort      /Applications/nextcloud.app/Contents/MacOS/nextcloud --logfile nextcloud.log
greg@Etna private %
`

**Comment by godfuture** [comment]
After standby I see a red icon with white X. NC client seems to be started, but sync in background is not working...even icons in windows explorer are green. I have to restart the client to bring it back.

All sync folders show the same message:
"Beim Öffnen eines Ordners ist ein Fehler aufgetreten. Der Zugriff auf das Netzwerk ist nicht gestattet."

Win10 x64
2.6.4

**Comment by amelekhin** [comment]
I can confirm that this issue is still reproducible on Windows 10 Home 2004 and Nextcloud 2.6.4. Every time my laptop hibernates or falls asleep, the Nextcloud client stops working. Only restarting the app makes it work properly. Unfortunately, I don't have experience in C++ to debug it by myself, so I hope someone can take a look at this issue. It's quite annoying and two years old already.

**Comment by weeheavy** [comment]
For me, the bug is reintroduced with 2.6.5, it was gone with the version before whatever that was. After hibernating, client shows a red cross, when clicking the tray icon it goes away/crashes. After starting the Nextcloud client again, it works until the next hibernation.

**Comment by gregholst** [comment]
I am on MacOS Catalina and since the new version it's the first time that NextCloud Desktop Client does not crash due to hibernation since I am using it (roughly 1.5 years). I guess the problem became "Windows-only"

**Comment by nightspotlight** [comment]
on version 3.0.1 for Windows the desktop client stopped crashing, but still after resuming computer from hibernation the client is unable to resume synchronization due to error "network connection is disabled" (Доступ в сеть отключён). Forcing synchronization doesn't help. how can I help debug this?

**Comment by q-wertz** [comment]
> I am on MacOS Catalina and since the new version it's the first time that NextCloud Desktop Client does not crash due to hibernation since I am using it (roughly 1.5 years). I guess the problem became "Windows-only"

On MacOS the crash happens again since the last update.
@gregholst can you confirm this?

**Comment by gregholst** [comment]
> > I am on MacOS Catalina and since the new version it's the first time that NextCloud Desktop Client does not crash due to hibernation since I am using it (roughly 1.5 years). I guess the problem became "Windows-only"
> 
> On MacOS the crash happens again with the newest version
> **NC Client Version:** 3.0.1git (build3355)
> 
> @gregholst can you confirm this?

Unfortunately, I have to confirm that within the newest version the problem was re-introduced on MacOS

**Comment by jameskimmel** [comment]
If have the same problem under Windows 10 with Version 3.0.1.

It is very annoying and should be very high on the priority list IMO! If offer nextcloud to non tech savvy friends and parents and it is hard to explain to them, that they should pause and resume every time they open up their laptops.

**Comment by kusmi** [comment]
Same problem on 3.0.2 Mac. When Laptop is put to sleep or has no network connection for a while (e.g. out-of-house or in locations with open WiFi networks which redirect all web-requests to a login/register page) - but then has internet connection again: It will often not recover and stops syncing altogether.

It often still shows the "checkmark" icon in the Dock (so a normal person would assume all is synced) - but only a restart of the nextcloud client will recover.

**Comment by robinedmunds** [comment]
I've started experiencing this issue on Windows 10, 1909. Nextcloud client v3.0.3.

On resuming from sleep, the client's tray icon is red, no syncing occurs but the app is responsive. 

Pause-resuming syncing does not fix the issue. Closing and relaunching the client restores normal function.

**Comment by barry-luijten** [comment]
Problem came back in 3.0.3 and is still present in client v3.1.0.
See also multiple reports in  #865

**Comment by heziegl** [comment]
workaround that worked for me was to use exe-installer instead of msi-installer.
see issue https://github.com/nextcloud/desktop/issues/482#issuecomment-751285146

**Comment by er-vin** [comment]
Isn't it duplicate of #482 indeed?

**Comment by MikeZuluUSA** [comment]
Problem came back with x64 client 3.1.1!
Close and restart the client solve the issue every time! 
Only find the msi-installer. Any workarounds?
THX

**Comment by MikeZuluUSA** [comment]
> Problem came back with x64 client 3.1.1!
> Close and restart the client solve the issue every time!
> Only find the msi-installer. Any workarounds?
> THX
Found it! https://github.com/nextcloud/desktop/issues/482
https://download.nextcloud.com/desktop/releases/Windows/

**Comment by cweakland** [comment]
I'm going to add a me too :D  However, 3.0.3 seems to be better than 3.1.1 for me.

**Comment by MikeZuluUSA** [comment]
> > Problem came back with x64 client 3.1.1!
> > Close and restart the client solve the issue every time!
> > Only find the msi-installer. Any workarounds?
> > THX
> > Found it! #482
> > https://download.nextcloud.com/desktop/releases/Windows/

Problem solved with the exe-installer file!

**Comment by tgwaste** [comment]
On mac M1 laptop every time I open the lid the client isn't even running.  I have to start it.  Would love a fix.

**Comment by PRhodan** [comment]
@MikeZuluUSA
> Problem solved with the exe-installer file!

What ist this for a shit, exe-installer file solve nothing!

**Comment by PRhodan** [comment]
Hello, it looks to me that the Windows 10 client sync. with Nextcloud-3.2.0-rc1-x64.msi works for me.
No crash after suspend mode and no "waiting for ...".
But actually NC 20.0.8.1 not supported - no problem for me:
![grafik](https://user-images.githubusercontent.com/37460114/111772067-ba859080-88ac-11eb-903e-9324f85cf6b2.png)

**Comment by rwilliamsit** [comment]
Latest client 3.2.0 windows x64 seems to be working for me.

**Comment by HieronymusCH** [comment]
I'm still having the network timeout error on my setting.
Doesn't go away, I have to exit and restart the client.

NextCloud Desktop Version 3.2.2 (Ubuntu)

Operating System: Kubuntu 20.10
KDE Plasma Version: 5.19.5
KDE Frameworks Version: 5.74.0
Qt Version: 5.14.2
Kernel Version: 5.8.0-54-generic
OS Type: 64-bit
Processors: 8 × Intel® Core™ i7-3632QM CPU @ 2.20GHz
Memory: 15.5 GiB of RAM
Graphics Processor: Mesa DRI Intel® HD Graphics 4000

**Comment by HieronymusCH** [comment]
Seems solved now.
Can anybody else confirm?

**Comment by mgallien** [comment]
@HieronymusCH @q-wertz thanks for the reply, I am closing it

**Comment by HieronymusCH** [comment]
Too soon it seems.
"Network error - time out" returned here this morning.

Is there any information I can provide to pinpoint this problem?
